src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -359,22 +379,7 @@ public void undo() throws CommandException {
                 filteredBills.setPredicate(history.getBillsPredicate(history.getBillsHistorySize() - 1));
                 history.deleteHealthContactHistory(history.getHealthContactHistorySize() - 1);
             } else {
-                try {
-                    history.deleteHealthContactHistory(history.getHealthContactHistorySize() - 1);
-                    history.updateRedoHealthContactHistory();
-                    history.updateRedoPatientsHistory();
-                    history.updateRedoAppointmentsHistory();
-                    history.updateRedoBillsHistory();
-                    setHealthContact(history.getHealthContactHistory(history.getHealthContactHistorySize() - 2));
-                    filteredPatients.setPredicate(history.getPatientsPredicate(history.getPatientsHistorySize() - 2));
-                    filteredAppointments.setPredicate(history
-                            .getAppointmentsPredicate(history.getAppointmentsHistorySize() - 2));
-                    filteredBills.setPredicate(history.getBillsPredicate(history.getBillsHistorySize() - 2));
-                    history.deleteHealthContactHistory(history.getHealthContactHistorySize() - 1);
-                    history.deleteHealthContactHistory(history.getHealthContactHistorySize() - 1);
-                } catch (IndexOutOfBoundsException e) {
-                    throw new CommandException("Undo cannot be done as there was no previous action");
-                }
+                throw new CommandException("Undo cannot be done as there was no previous change in data");
```

Comment:
Can pull my PR first before updating? If not my undo/redo functionality is deleted again